primap2/_downscale.py

```diff
@@ -104,6 +111,32 @@ def downscale_timeseries(
                     "To continue regardless, set check_consistency=False."
                 )
 
+        # treat zeros
+        basket_both_zero = basket_da.where((basket_da == 0) & (basket_sum == 0))
+        basket_sum_zero = basket_da.where((basket_da != 0) & (basket_sum == 0))
+        if not basket_sum_zero.isnull().all():
+            # this can only happen if check consistency = False,
+            # so we could also remove it and say whoever switches it of has
+            # to deal with the consequences
+            error_message = generate_error_message(basket_sum_zero)
+            raise ValueError(f"pr.downscale_timeseries {error_message}")
+
+        any_nonzero = basket_sum.where(basket_sum != 0).notnull().any()
+
+        # treat the case where all data is zero or NaN
+        if (not basket_both_zero.isnull().all()) & (not any_nonzero):
+            unit_content = str(basket_contents_da.pint.units)
+            basket_da_squeeze = basket_da.drop(dim)
+            basket_contents_da = basket_contents_da.where(
+                (basket_da_squeeze != 0) | (basket_sum != 0), 1 * ureg(unit_content)
+            )
```

Comment:
```suggestion
            units = basket_contents_da.pint.units
            basket_da_squeeze = basket_da.drop(dim)
            basket_contents_da = basket_contents_da.where(
                (basket_da_squeeze != 0) | (basket_sum != 0), 1 * units
            )
```
Is there a reason why this wouldn't work? Is the conversion-to-string-and-back necessary?

Reply:
It makes no sense when looking at it. I had some problem with the unit conversion and it might just be a result of fiddling around until I had something that worked. I remember that a bit later I saw unit treatment in another function and thought that it looked more elegant than what I did. So probably it's just bad code.